// trello.js/out/clients/trelloClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrelloClient = void 0;
const baseClient_1 = require("./baseClient");
const api_1 = require("../api");
class TrelloClient extends baseClient_1.BaseClient {
    constructor() {
        super(...arguments);
        this.actions = new api_1.Actions(this);
        this.applications = new api_1.Applications(this);
        this.batch = new api_1.Batch(this);
        this.boards = new api_1.Boards(this);
        this.cards = new api_1.Cards(this);
        this.checklists = new api_1.Checklists(this);
        this.customFields = new api_1.CustomFields(this);
        this.emoji = new api_1.Emoji(this);
        this.enterprises = new api_1.Enterprises(this);
        this.labels = new api_1.Labels(this);
        this.lists = new api_1.Lists(this);
        this.members = new api_1.Members(this);
        this.notifications = new api_1.Notifications(this);
        this.organizations = new api_1.Organizations(this);
        this.plugins = new api_1.Plugins(this);
        this.search = new api_1.Search(this);
        this.tokens = new api_1.Tokens(this);
        this.webhooks = new api_1.Webhooks(this);
    }
}
exports.TrelloClient = TrelloClient;
